File: app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.document import DocumentStatus


class DocumentRead(BaseModel):
    id: int
    filename: str
    content_type: str
    storage_path: str
    status: DocumentStatus
    created_at: datetime
    pages_count: int | None = None
    words_count: int | None = None
    chunks_count: int | None = None
    language: str | None = None
    document_type: str | None = None
    document_type_confidence: float | None = None

    model_config = ConfigDict(from_attributes=True)


class DocumentSummaryResponse(BaseModel):
    document_id: int
    filename: str
    summary: str

File: web/src/types/document.ts
export type DocumentRead = {
  id: number;
  filename: string;
  content_type: string;
  storage_path: string;
  status: string;
  created_at: string;

  pages_count?: number | null;
  words_count?: number | null;
  chunks_count?: number | null;
  language?: string | null;

  document_type?: string | null;
  document_type_confidence?: number | null;
};